core/src/test/scala/io/finch/request/ReaderAsServiceSpec.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2014, by Vladimir Kostyukov and Contributors.
+ *
+ * This file is a part of a Finch library that may be found at
+ *
+ *      https://github.com/finagle/finch
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributor(s):
+ * Jens Halm
+ */
+package io.finch.request
+
+import org.scalatest.{Matchers, FlatSpec}
+import com.twitter.util.{Future, Await}
+import com.twitter.finagle.httpx.Request
+import com.twitter.finagle.Service
+import io.finch.HttpRequest
+
+class ReaderAsServiceSpec extends FlatSpec with Matchers {
+
+  
+  "A RequestReader" should "be usable as a Finagle Service" in {
+    case class MyRequest(request: HttpRequest)
+    
+    implicit def extractRequest(req: MyRequest): HttpRequest = req.request
+    
+    val request: MyRequest = MyRequest(Request(("foo", "5")))
+    val reader: RequestReader[Int] = RequiredParam("foo").as[Int]
+    val service: Service[MyRequest, Int] = reader.asService[MyRequest]
```

Comment:
We have [a similar conversion|https://github.com/finagle/finch/blob/master/core/src/main/scala/io/finch/package.scala#L203] for `Future[Rep] -> Service[Req, Rep]`. So, the compile will figure out the `Req` type in most of the cases. In our case, we have an explicit type `Service[MyRequest, Int]`. In case of service composition via `!` it would definitely work when `reader` is not the first service in the chain. We should probably experiment with it in tests.